test/dummy/test/controllers/protected_resources_controller_test.rb

```diff
@@ -15,6 +15,12 @@ def authenticate token: @token
     assert_response :unauthorized
   end
 
+  test "responds with unauthorized with empty token in header" do
```

Comment:
If you follow my tip in the setter/getter thing, this test won't be necessary anymore. Look, we won't simply fail the request with unauthorized response. Instead, it will prevent the host app from loading until a valid token be properly set. I think this way makes the issue explicit and help the devs to fix it faster...